src/components/PostDetails.tsx

```diff
@@ -1,117 +1,133 @@
-          </p>
+        updatePostComments.splice(index, 1);
+        setPostComments(updatePostComments);
+      })
+      .catch(() => setHasError(true));
+  };
 
-          <p className="title is-4">Comments:</p>
+  useEffect(() => {
+    if (openPost) {
+      setLoading(true);
+      setHasError(false);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+      getComment(openPost.id)
+        .then(PostComments => setPostComments(PostComments))
+        .catch(() => setHasError(true))
+        .finally(() => setLoading(false));
+    }
+  }, [openPost]);
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a
-                href="mailto:misha@mate.academy"
-                data-cy="CommentAuthor"
-              >
-                Misha Hrynko
-              </a>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
+        <div className="block">
+          <h2 data-cy="PostTitle">{`#${openPost?.id}: ${openPost?.title}`}</h2>
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div
-              className="message-body"
-              data-cy="CommentBody"
-            >
-              One more comment
-            </div>
-          </article>
+          <p data-cy="PostBody">{openPost?.body}</p>
+        </div>
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a
-                href="mailto:misha@mate.academy"
-                data-cy="CommentAuthor"
-              >
-                Misha Hrynko
-              </a>
+        {!hasError && (
+          <div className="block">
+            {loading ? (
+              <Loader />
+            ) : (
+              <>
+                {postComments.length === 0 ? (
+                  <p className="title is-4" data-cy="NoCommentsMessage">
+                    No comments yet
+                  </p>
+                ) : (
+                  <>
+                    <p className="title is-4">Comments:</p>
+                    {postComments.map((com) => (
```

Comment:
```suggestion
                    {postComments.map((comment) => (
```
do not use short forms for variable names